e2e/cypress/integration/messaging/pinned_parent_post_spec.js

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @messaging
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('Messaging', () => {
+    let testTeam;
+    let testChannel;
+    let receiver;
+    let sender;
+    let lastPostId;
+
+    before(() => {
+        // # Login as test user
+        cy.apiInitSetup().then(({team, channel, user}) => {
+            receiver = user;
+            testTeam = team;
+            testChannel = channel;
+
+            cy.apiCreateUser().then(({user: user1}) => {
+                sender = user1;
+                cy.apiAddUserToTeam(testTeam.id, sender.id).then(() => {
+                    cy.apiAddUserToChannel(testChannel.id, sender.id);
+                });
+            });
+
+            cy.apiLogin(receiver);
+
+            // # Visit a test channel and post a message
+            cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+            cy.postMessage('Hello');
+
+            // # Assign lastPostId variable to the id of the last post
+            cy.getLastPostId().then((postId) => {
+                lastPostId = postId;
+            });
```

Comment:
Probably best to move this into the test block itself.

Reply:
I moved it to the test block as you suggested. However, after I moved it to test block, the `lastPostId = postId;` variable assignment expression did not work as intended, so I had to wrap all the commands that use `postId` within the `cy.getLastPostId` function.

I'm not sure why the variable reassignment did not work, have you encountered this issue by any chance? I tried creating a test variable `a` in the global scope, reassigned that variable in the test block and console.log it and it was working as intended, but reassigning a variable to the value of the postId returned from the promise doesn't work in the test block, but it does in the`before` block.

Reply:
I think the assignment happened first on the test block before going into the before hook. That's why the value is getting is the initial value of undefined. But it's just my assumption as I'm getting similar issue in the past.

Reply:
But the one you implemented is the one I'm thinking actually. Wrapping everything inside the `cy.getLastPostId`.

Reply:
Ah, thanks for the explanation Saturn!